.github/report-flaky-tests/index.js

```diff
@@ -0,0 +1,249 @@
+};
+
+const metaData = {
+	render: ( json ) => `<!-- __META_DATA__:${ JSON.stringify( json ) } -->`,
+	get: ( str ) => {
+		const matched = str.match( /<!-- __META_DATA__:(.*) -->/ );
+		if ( matched ) {
+			return JSON.parse( matched[ 1 ] );
+		}
+	},
+};
+
+( async function run() {
+	const token = core.getInput( 'repo-token', { required: true } );
+	const label = core.getInput( 'label', { required: true } );
+	const artifactName = core.getInput( 'artifact-name', { required: true } );
+
+	const octokit = github.getOctokit( token );
+
+	const flakyTests = await downloadReportFromArtifact(
+		octokit,
+		artifactName
+	);
+
+	if ( ! flakyTests ) {
+		return;
+	}
+
+	const { data: issues } = await octokit.rest.issues.listForRepo( {
+		...github.context.repo,
+		labels: label,
+		per_page: 100,
+	} );
+
+	for ( const flakyTestKey in flakyTests ) {
+		const [ testPath, testTitle ] = getTestPathAndTitle( flakyTestKey );
+		const issueTitle = getIssueTitle( testTitle );
+		const reportedIssue = issues.find(
+			( issue ) => issue.title === issueTitle
+		);
+		const testResults = flakyTests[ flakyTestKey ];
+		const isTrunk = getHeadBranch() === 'trunk';
+
+		if ( reportedIssue ) {
+			let body = reportedIssue.body;
+			const meta = metaData.get( body );
+
+			if ( isTrunk ) {
+				const headCommit = github.context.sha;
+				const baseCommit = meta.baseCommit || github.context.sha;
+
+				let totalCommits = 0;
+				try {
+					const { data } = await octokit.rest.repos.compareCommits( {
+						...github.context.repo,
+						base: baseCommit,
+						head: headCommit,
+						per_page: 1,
+					} );
+
+					meta.failedTimes += testResults.length;
+					meta.totalCommits = data.total_commits + 1;
+				} catch ( err ) {
+					// It might a forced push or a deleted commit,
+					// we have to treat the current commit as the base commit.
+					meta.baseCommit = headCommit;
+					meta.failedTimes = testResults.length;
+					meta.totalCommits = 1;
+				}
+			}
+
+			body =
+				renderIssueDescription( { meta, testTitle, testPath } ) +
+				body.slice(
+					body.indexOf( TEST_RESULTS_LIST.open ),
+					body.indexOf( TEST_RESULTS_LIST.close )
+				) +
+				[
+					renderTestErrorMessage( { testPath, testResults } ),
+					TEST_RESULTS_LIST.close,
+				].join( '\n' );
+
+			await octokit.rest.issues.update( {
+				...github.context.repo,
+				issue_number: reportedIssue.number,
+				body,
+			} );
+		} else {
+			const meta = isTrunk
+				? {
+						failedTimes: testResults.length,
+						totalCommits: 1,
+						baseCommit: github.context.sha,
+				  }
+				: {
+						failedTimes: 0,
+						totalCommits: 0,
+				  };
+
+			await octokit.rest.issues.create( {
+				...github.context.repo,
+				title: issueTitle,
+				body: renderIssueBody( {
+					meta,
+					testTitle,
+					testPath,
+					testResults,
+				} ),
+				labels: [ label ],
+				// TODO: Maybe we can combine git blame to automatically assign issue to the original author?
```

Comment:
On the other hand, if original author has gone on holiday, or is otherwise absent, having the issue assigned might just stop other folks from picking it up. Maybe best leave it as is for now.

Reply:
I would like to stay away from auto assigning for now.